### DISCUSSION TITLE
How does Dogecoin relay transaction

### DISCUSSION THREAD

**Discussion by user404dev** [dogecoin/dogecoin#2387] [Q&A]
Hello,
This is just a question not a issue or something like that I just need to know this for my knowledge...

I was wondering that, I know that every node on the network has a copy of a Dogecoin blockchain. But how do they update it? Like for example let’s say Dogecoin stored my transactions on my local machine. But how does other nodes get this transaction if it’s not hosted online and only stored locally? I’m searching the internet for abkut 3 days and I didn’t found the answer. The only thing I found was that the blockchain is stored locally. 

And another question is like how are all the address stored? Like what if I sent my amount in a wrong address how will the system know if it’s right or wrong

I hope you guys will answer the answer to the questions for me

**Comment by patricklodder** [comment]
Hi! It works with peer-to-peer relay, see https://en.wikipedia.org/wiki/Bitcoin_network#Process

**Comment by AbcSxyZ** [comment]
[Gossip protocol](https://academy.bit2me.com/en/what-is-gossip-protocol/) :) ?

> And another question is like how are all the address stored? 

To help you, I think to understand how private key are generated would be great. It's like choosing a [random number](https://books.google.fr/books?id=MpwnDwAAQBAJ&pg=PA58&lpg=PA58&dq=generate+new+bitcoin+wallet+%22pick%22+%22random%22&source=bl&ots=wR5pqnx_z2&sig=ACfU3U3rTlzI72f1YxC88BDH47RshZvpKQ&hl=fr&sa=X&ved=2ahUKEwjY37aCoN3xAhW2A2MBHXilCZUQ6AEwB3oECAoQAw#v=onepage&q=generate%20new%20bitcoin%20wallet%20%22pick%22%20%22random%22&f=false).

By logical deduction, the Dogecoin network do not know the existence of a public key until there is any transaction. You create first your private key and getting a public key derivated from it, but no information are stored on the blockchain at first, no [UTXO](https://en.wikipedia.org/wiki/Unspent_transaction_output) available in the ledger for this wallet.

I think you can send some money to a "non-picked" wallet, where you need a non-picked private key to unlock this fund.
